frame/staking/src/pallet/mod.rs

```diff
@@ -433,6 +437,17 @@ pub mod pallet {
 	#[pallet::getter(fn current_planned_session)]
 	pub type CurrentPlannedSession<T> = StorageValue<_, SessionIndex, ValueQuery>;
 
+	/// Indices of validators that have offended in the current era and whether they are currently
+	/// disabled.
+	///
+	/// This value should be a superset of disabled validators since not all offences lead to the
+	/// validator being disabled (if there was no slash). This is needed to track the percentage
+	/// of validators that have offended in the current era, ensuring a new era is forced if
+	/// `OffendingValidatorsThreshold` is reached. The set is cleared when the era ends.
+	#[pallet::storage]
+	#[pallet::getter(fn offending_validators)]
+	pub type OffendingValidators<T: Config> = StorageValue<_, Vec<(u32, bool)>, ValueQuery>;
```

Comment:
```suggestion
	pub type OffendingValidators<T: Config> = StorageValue<_, BoundedVec<(u32, bool)>, ValueQuery>;
```

Can you already make this a bounded vec? It will be something we will need to update and break in the future otherwise.

Reply:
note you will also need to add a bounding condition here

Reply:
The bounding condition would be the maximum number of validators but that's dynamic (`MaxValidatorsCount` storage item), so I'm not sure how I could go about using it here.

Reply:
We should have a static upper limit which even bounds the dynamic limit, and then if we need to go higher, we need to do a runtime upgrade.

Unfortunately, I think this is the only way...

Reply:
yeah probably using a weak bonded vec is the easiest. I see we are going to have MaxAuthorities in pallet-authority-discovery, pallet-aura. I guess we can have same in pallet-staking. And use this max to bound this storage.

Anyway we can do in a follow-up